eclair-core/src/main/scala/fr/acinq/eclair/channel/Helpers.scala

```diff
@@ -990,6 +975,9 @@ object Helpers {
       } else if (nextRemoteCommit_opt.map(_.txid).contains(tx.txid)) {
         // their last commitment got confirmed, so no htlcs will be overridden, they will timeout or be fulfilled on chain
         Set.empty
+      } else if (d.revokedCommitPublished.map(_.commitTx.txid).contains(tx.txid)) {
+        // a revoked commitment got confirmed: we will claim its outputs, but we also need to fail htlcs that are pending in the latest commitment
+        nextRemoteCommit_opt.getOrElse(remoteCommit).spec.htlcs.collect(incoming)
```

Comment:
>  there should be some comparisons with existing local

Why? It feels to me that it's exactly the same case as elsewhere: we want to fail every pending signed HTLC (they will never be settled since the remote published a revoked commitment, which means that all the content of this channel can be ignored). All our pending HTLCs have been overridden by this revoked commitment, so the naming feels correct to me.

Reply:
> since we are going to claim their output as a punishment we will never get the preimage and may as well consider them failed in the context of relaying htlcs

That's part of the intuition indeed. Once the remote has published a revoked tx, every HTLC in the current commit tx can be safely failed upstream. There should be no way the downstream peer can claim these HTLCs, we should claim them all via penalty txs (but without learning the preimage).

Reply:
My point is that the original meaning of this method is to identify `htlcs that will never reach the blockchain`. In the case of a revoked close, those htlcs will hit the chain, but we decide to fail them for other reason. That's why I'm proposing to add the second bullet point above as comment. The first bullet point probably can be ignored.

Reply:
Got it, it's indeed a weird case because these HTLCs will hit the chain, but not in a meaningful way - this doesn't count as normal htlc settlement. So it could be interpreted in different ways, I choose to interpret it as a punishment to the remote, but not as HTLCs being settled.

I'll update the comment.

Reply:
Done in https://github.com/ACINQ/eclair/pull/1630/commits/1ad2c80a7078dd7f321cecc5e82c4f86ead7c91f